src/App.tsx

```diff
@@ -1,61 +1,109 @@
 import './App.scss';
+import { TodoList } from './components/TodoList';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import React, { useState } from 'react';
+import { Todo } from './types/Todo';
+
+const todos = todosFromServer.map(todo => {
+  return {
+    ...todo,
+    user: usersFromServer.find(user => user.id === todo.userId) || null,
+  };
+});
 
 export const App = () => {
+  const [todo, setToDo] = useState<Todo[]>(todos);
+
+  const [selectedUserId, setSelectedUserId] = useState(0);
+  const [title, setTitle] = useState('');
+
+  const [hasTitleError, setHasTitleError] = useState(false);
+  const [hasUserIdError, setHasUserIdError] = useState(false);
+
+  function getNewTodoId(todosList: Todo[]) {
+    const maxId = Math.max(...todosList.map(t => t.id));
+
+    return maxId + 1;
+  }
+
+  const handleChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setSelectedUserId(+event.target.value);
+    setHasUserIdError(false);
+  };
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { value } = event.target;
+
+    setHasTitleError(false);
+    setTitle(value);
+  };
+
+  const handelSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    setHasTitleError(!title);
+    setHasUserIdError(!selectedUserId);
+    if (!title || !selectedUserId) {
```

Comment:
```suggestion
    if (!title.trim() || !selectedUserId) {
```

Reply:
@anastasiiavorobiova thank you for advices!)